app/jinja_filters.py

```diff
@@ -82,8 +82,21 @@ def format_address_list(user_entered_address=None, metadata_address=None):
     return address_list
 
 
-def format_unit(unit, value=''):
-    return units.format_unit(value=value, measurement_unit=unit, length='short', locale=DEFAULT_LOCALE)
+def format_unit(unit, value='', length='short'):
+    return units.format_unit(value=value, measurement_unit=unit, length=length, locale=DEFAULT_LOCALE)
+
+
+def format_unit_input_label(unit, length='short', use_plural_for_label=False):
+    """
+    This function is used to only get the unit of measurement text for answer fields.
+
+    :param (str) unit unit of measurement
+    :param (str) length length of unit  can be one of short/long/narrow
+    :param (bool) use_plural_for_label if true, will return the plural version of the unit.
+    """
+    if use_plural_for_label:
+        return units.format_unit(value=2, measurement_unit=unit, length=length, locale=DEFAULT_LOCALE).replace('2 ', '')
+    return units.format_unit(value='', measurement_unit=unit, length=length, locale=DEFAULT_LOCALE).replace(' ', '', 1)
```

Comment:
Does the current implementation have a space before the unit then?

Reply:
Yea, we just didn't notice it because it gets ignored when it's rendered.
There weren't any unit tests around using `format_unit` for labels before, but I've added them now so little surprises like this shouldn't happen again

Reply:
We should probably just use trim then?